pkg/cloud-provider/cloudapi/aws/aws_services.go

```diff
@@ -54,29 +53,32 @@ type awsServicesHelperImpl struct{}
 // newServiceSdkConfigProvider returns config to create aws services clients.
 func (h *awsServicesHelperImpl) newServiceSdkConfigProvider(accConfig *awsAccountConfig) (awsServiceClientCreateInterface, error) {
 	var creds *credentials.Credentials
-
+	var err error
 	if len(accConfig.RoleArn) != 0 {
-		// use role base access if role provided
-		// new session using worker node role, it should have AssumeRole permissions to the Customer's role ARN resource
-		sess, err := session.NewSession(&aws.Config{
-			Region:                        &accConfig.region,
-			CredentialsChainVerboseErrors: aws.Bool(true),
-		})
+		var sess *session.Session
+		// If credentials are specified too, create a session with these credentials.
+		if len(accConfig.AccessKeyID) != 0 && len(accConfig.AccessKeySecret) != 0 {
+			tempCreds := credentials.NewStaticCredentials(accConfig.AccessKeyID, accConfig.AccessKeySecret, "")
+			sess, err = session.NewSession(&aws.Config{
+				Region:                        &accConfig.region,
+				Credentials:                   tempCreds,
+				CredentialsChainVerboseErrors: aws.Bool(true),
+			})
+
+		} else {
+			// use role base access if role provided
+			// new session using worker node role, it should have AssumeRole permissions to the Customer's role ARN resource
+			sess, err = session.NewSession(&aws.Config{
+				Region:                        &accConfig.region,
+				CredentialsChainVerboseErrors: aws.Bool(true),
+			})
+		}
 		if err != nil {
 			return nil, fmt.Errorf("unable to initialize AWS session: %v", err)
 		}
 
 		// configure to assume customer role and retrieve temporary credentials
-		externalID := &accConfig.ExternalID
-		if len(accConfig.ExternalID) == 0 {
-			externalID = nil
```

Comment:
Do you want to remove it from the AwsAccountCredential as well?

Reply:
will remove at part of CRD update